cli/src/utils.rs

```diff
@@ -80,24 +80,28 @@ impl PlatformSelector {
                 // defaults based on previous selections if previous selections are not empty
                 // otherwise, selects all
                 let all_ios_archs = IosArch::all_strings();
-                let defaults: Vec<bool> = if config.ios.is_empty() {
+                let defaults: Vec<bool> = if config.ios.is_none() {
                     vec![true; all_ios_archs.len()]
                 } else {
                     all_ios_archs
                         .iter()
-                        .map(|&acrch| config.ios.contains(acrch))
+                        .map(|&acrch| config.ios.as_ref().unwrap().contains(acrch))
                         .collect()
                 };
 
                 // clear previous selections before update
-                config.ios.clear();
+                if let Some(ref mut ios) = config.ios {
+                    ios.clear();
+                }
 
                 let sel = Self::select_multi_archs(p.as_str(), &all_ios_archs, defaults);
                 let sel_str = sel
                     .iter()
                     .map(|&i| {
                         let arch = IosArch::from_idx(i).as_str().to_string();
-                        config.ios.insert(arch.clone());
+                        if let Some(ref mut ios) = config.ios {
```

Comment:
the same here, we don't need `config.ios`, just update `ios` directly is enough.

Reply:
the same as `android` below

Reply:
[029aaec](https://github.com/zkmopro/mopro/pull/417/commits/029aaec3858a5897fbbc747d4efea1752807973e)

Reply:
`config.ios` was re-created above (`config.ios = Some(HashSet::new());`) so we don't need check if it's `Some`.

Reply:
[76e1642](https://github.com/zkmopro/mopro/pull/417/commits/76e164275fd080011eca11d337734fa621d7c357)